### DISCUSSION TITLE
Single stepping in the debugger highlights the wrong line.

### DISCUSSION THREAD

**Discussion by GethBo101** [maziac/DeZog#115] [Q&A] [answered]
Hi Folks,

I'm single stepping a program and I can see that the highlighted next instruction in the source code and the next instruction in the disassembly are not in sync. In the below screen shot the line ld B,05H is highlighted. But the registers screen has pc = 9002h and the next instruction at that address is 'or a' on the line below the highlighted line.

Stepping the instruction does indeed run the 'or a' instruction, and then moves the highlighted line down.

Anyone know how to fix this ?

![Screenshot 2023-08-28 at 15 31 15](https://github.com/maziac/DeZog/assets/97247221/8d439a45-e1b5-448f-b808-36cfe6f2084f)

**Comment by maziac** [comment, marked as answer]
The disassembly on the left is always right.
For some reason the references in the sld file do not seem to be uptodate.
This can happen if you changed sources but might have forgotten to compile or maybe the launch.json is not using the sld file that is being compiled.

If that all is not the case I can have a look if you zip your project and upload it here.
(Together with the .vscode/launch.json and the sld file.)

**Reply by GethBo101** [reply]
Hi Maziac,

You were right, that was exactly the problem, the sld file was not up to date. 

I had been fiddling with the build task and changed the name of it.

It's working as expecting now, thank you for your help with this.

**Comment by maziac** [comment]
Great that it is working now.
